gneiss/regression/_mixedlm.py

```diff
@@ -0,0 +1,270 @@
+                                                              tree)
+    ilr_table, basis = _to_balances(table, tree)
+    data = pd.merge(ilr_table, metadata, left_index=True, right_index=True)
+    submodels = []
+    for b in ilr_table.columns:
+        # mixed effects code is obtained here:
+        # http://stackoverflow.com/a/22439820/1167475
+        stats_formula = '%s ~ %s' % (b, formula)
+        mdf = smf.mixedlm(stats_formula, data=data,
+                          groups=data[groups],
+                          **kwargs)
+        submodels.append(mdf)
+
+    return LMEModel(submodels, basis=basis,
+                    balances=ilr_table, tree=tree)
+
+
+class LMEModel(RegressionModel):
+    def __init__(self, *args, **kwargs):
+        """
+        Summary object for storing linear mixed effects results.
+
+        A `LMEModel` object stores information about the
+        individual balances used in the regression, the coefficients,
+        residuals. This object can be used to perform predictions.
+        In addition, summary statistics such as the coefficient
+        of determination for the overall fit can be calculated.
+
+
+        Parameters
+        ----------
+        submodels : list of statsmodels objects
+            List of statsmodels result objects.
+        basis : pd.DataFrame
+            Orthonormal basis in the Aitchison simplex.
+            Row names correspond to the leafs of the tree
+            and the column names correspond to the internal nodes
+            in the tree. If this is not specified, then `project` cannot
+            be enabled in `coefficients` or `predict`.
+        tree : skbio.TreeNode
+            Bifurcating tree that defines `basis`.
+        balances : pd.DataFrame
+            A table of balances where samples are rows and
+            balances are columns.  These balances were calculated
+            using `tree`.
+        """
+        super().__init__(*args, **kwargs)
+
+    def fit(self, **kwargs):
+        """ Fit the model
+
+        Parameters
+        ---------
+        regularized : bool
+            Indicates if the fixed effects parameters are penalized.
+            This is useful when there are many more parameters than
+            data points.
+        """
+        for s in self.submodels:
+            # assumes that the underlying submodels have implemented `fit`.
+            # TODO: Add regularized fit
+            # if regularized:
+            #    m = s.fit_regularized(**kwargs)
+            m = s.fit(**kwargs)
+            self.results.append(m)
+
+    def summary(self, title=None, yname=None, xname=None, head=None):
+        """ Summarize the Ordinary Least Squares Regression Results.
+
+        Parameters
+        ----------
+        title : string, optional
+            Title for the top table. If not None, then this replaces the
+            default title
+        yname : string, optional
```

Comment:
is yname the label for the y axis? If yes, you might want to rename it into ylabel to be more consistent to e.g. matplotlib